proof-defns/proof_defns.tex

```diff
@@ -0,0 +1,438 @@
+
+\questionc{We should decide on whether we want to keep a vector approach or a multiset approach:}
+\begin{definition}[Histogram notation, multiset version]
+\label{defn:histogram}
+    Let $h_x: \mathcal{X} \rightarrow \mathbb{N}$ be the histogram of a multiset $x \in \MultiSet(\mathcal{X} )$ for some domain $\mathcal{X}$. That is, $h_x(z)$ denotes the number of occurrences of $z \in \mathcal{X}$ in multiset $x$ (with multiplicities).
+\end{definition}
+
+%\todo{Change to vector domain notation. For vector $v$, $h_v(z)$ denotes the number of occurrences of $z$ in $\MultiSet(v)$.} \todo{Specify types: z?}
+
+\begin{definition}[Histogram notation, vector version]
+    For any vector $v$ of elements of a domain \texttt{D}, $h_v$ denotes the histogram of $v$. That is, for every element $z$ of type \texttt{T}, $h_v(z)$ denotes the number of occurrences of $z \in \texttt{D}$ in the entries of vector $v$ (with multiplicities).
+\end{definition}
+
+\todo{Say there is a bijection between multisets and histograms.}
+
+%Given a vector $v$ in some vector domain, $h_v(z)$ denotes the number of occurrences of $z$ in $\MultiSet(v)$.
+
+\section{Transformations \& Stability relation}
+%\silvia{Transformations here? Section for them?}
+\begin{definition}[Transformation]
+    A transformation $T$ is a \textit{deterministic} mapping from arbitrary data types of derived data values to arbitrary data types of derived data values. In Rust, a transformation is specified by the following attributes: input domain, output domain, function, input metric, output metric, and stability relation. 
+\end{definition}
+
+See \href{https://www.overleaf.com/project/60d215bf90b337ac02200a99}{``List of definitions used in the pseudocode"} for further details of the pseudocode specification of a transformation.
+
+%\silvia{Not c in R}
+\begin{definition}[Stability parameter]\label{def:c}
+    For some value of $c$, we say that a transformation $T$ is $c$\textit{-stable} if for all $x, x'$ in the input domain $\X$, and for input metric $d_{\X}$ and output metric $d_{\Y}$,
+    \begin{equation}
+        d_{\mathcal{Y}}(T(x), T(x')) \leq c \cdot d_{\mathcal{X}}(x, x').
+    \end{equation}
+    We say that $c$ is the \textit{stability parameter} of $T$.
+\end{definition}
+
+\silvia{Add a sentence saying that not all stability relations can be expressed only with $c$ (linear relationship).}
+
+In the Rust code, the stability parameter $c$ (which in the case of clamping is equal to~1) gets wrapped up inside of the stability relation property, and the end user can test it empirically. \todo{Explain better -- but perhaps in the general guidelines doc?}
+
+\begin{definition}[Stability relation]
+    The \textit{stability relation}, denoted $\Relation(\din, \dout)$, is a boolean function which takes as input some $\din, \dout$ appropriately quantified and returns \texttt{True} if and only if the relation $\dout \geq c \cdot \din$ for some specified value of $c$. 
```

Comment:
This can be made more general. This definition is specific to c-stable relations.

Reply:
We changed it to "linear stability relation" and added a comment on the generality -- would you add a definition for the general version?

Reply:
Update: we added a definition for the general version